### ISSUE TITLE
DELETE Application Role - Fails with 500 Internal Server Error

### ISSUE THREAD

**Issue by SlothFriend** [FusionAuth/fusionauth-issues#717]
### Description 
Calling the Delete Application Role endpoint returns a 500 error. Following the documentation here: https://fusionauth.io/docs/v1/tech/apis/applications#delete-an-application-role

### Steps to reproduce
1. Make an HTTP call to `DELETE https://{{HOST}}/api/application/{{APP_ID}}/role/{{ROLE_ID}}`
2. Use an API Key for auth (maybe unrelated, but that's what we are using)
3. See 500 Internal Server Error response and payload:
```
{
    "generalErrors": [
        {
            "code": "[Exception]",
            "message": "FusionAuth encountered an unexpected error. Please contact support for assistance."
        }
    ]
}
```

### Expected behavior
As far as I'm aware the call should be succeeding as documented here: https://fusionauth.io/docs/v1/tech/apis/applications#delete-an-application-role

If in fact something is wrong with the request, a status other than 500 should be returned with details about what was wrong with the request so that users can fix things and try again.

### Platform
(Please complete the following information)
 - Device: Desktop
 - OS: macOS
 - Experienced with NodeJS + Axios lib and also in Postman

**Comment by mooreds** [comment]
Hmmm. I'm unable to recreate. I tried both a role that a user had and one without.

Here's my curl script:

```
API_KEY=...

curl -XDELETE  -H "Authorization: $API_KEY" 'http://localhost:9011/api/application/a5d84462-193d-4c0d-9f20-d05a8a584710/role/4f275011-7e97-4858-9507-7fe7aabcd396' 
```

Where `a5d84462-193d-4c0d-9f20-d05a8a584710` is my app id and `4f275011-7e97-4858-9507-7fe7aabcd396` is my role id.

A few questions:
* What do the server logs show?
* Anything in particular special about the role? (default, superadmin?)
* What version of FusionAuth are you running?

One thing that caught me was that the name of the role is not the same as the id. Are you pulling the id of the role from the application configuration?

**Comment by mooreds** [comment]
@SlothFriend is this still an issue? Can you help with replication steps?

**Comment by robjanssen** [comment]
Hi! I think I found something that may help.

## Deleting an Application role fails with a 400 error

### Description 
Deleting an application roles causes an error.

### Affects versions
1.17.5

### Steps to reproduce
Steps to reproduce the behavior:
0. Log into the FusionAuth installation
1. Go to Applications and create a new application
2. Create a new Role for this application (no Super role, no default)
3. Go to Groups 
4. Create a new group
5. Assign the Role for that application you created to the Group
6. Go back to Applications
7. Attempt to delete the Role.
8. A "Confirm delete" pop-up appears, but on clicking Submit, there's a "FusionAuth Encountered an unexpected error" notification.

### Expected behavior
The role should have been deleted, even though it was linked to a group.

### Screenshots
No screenshot but a text dump from Docker:

Caused by: org.postgresql.util.PSQLException: ERROR: update or delete on table "application_roles" violates foreign key constraint "group_application_roles_fk_2" on table "group_application_roles"
Detail: Key (id)=(7c853795-062f-4c20-81c3-53f40a45e4e5) is still referenced from table "group_application_roles".

### Platform
(Please complete the following information)
 - Device: Desktop
 - OS: Docker Desktop on Windows 10 64-bit
 - Chrome 84.0.4147.105
 - Database: PostgreSQL

### Additional context
-

**Comment by robotdan** [comment]
This may be the same issue described here: https://github.com/FusionAuth/fusionauth-issues/issues/831 It was fixed in version 1.19.5. The root cause was a FK violation, so the recreate step is to add an application role to a group, and then try to delete the application role. 

I'll close out for now, assuming it is the same issue. If you can still recreate after 1.19.5, please re-open.
